hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootControllerTest.java

```diff
@@ -362,13 +370,77 @@ public void tryToFinishAnUpdateProcessAfterItHasBeenFinished() throws Exception
+    @ExpectEvents({ @Expect(type = TargetCreatedEvent.class, count = 1),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 2),
+            @Expect(type = TargetAssignDistributionSetEvent.class, count = 2),
+            @Expect(type = ActionCreatedEvent.class, count = 2), @Expect(type = ActionUpdatedEvent.class, count = 2),
+            @Expect(type = TargetUpdatedEvent.class, count = 6), @Expect(type = TargetPollEvent.class, count = 4),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 6),
+            @Expect(type = TargetAttributesRequestedEvent.class, count = 1) })
+    public void testAttributeUpdateRequestSendingAfterSuccessfulDeployment() throws Exception {
+        final String targetControllerId = "922";
+        final DistributionSet ds1 = testdataFactory.createDistributionSet("1");
+        final DistributionSet ds2 = testdataFactory.createDistributionSet("2");
+        Target savedTarget = testdataFactory.createTarget(targetControllerId);
+        final Map<String, String> attributes = Collections.singletonMap("AttributeKey", "AttributeVlue");
+        JsonBuilder.configData(targetControllerId, attributes, ExecutionStatus.CLOSED.getName());
+        assertAttributesRequestStatus(targetControllerId, true);
+
+        mvc.perform(put("/{tenant}/controller/v1/{controllerId}/configData", tenantAware.getCurrentTenant(),
+                targetControllerId).content(
+                        JsonBuilder.configData(targetControllerId, attributes, ExecutionStatus.CLOSED.getName()))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        assertAttributesRequestStatus(targetControllerId, false);
+
+        savedTarget = assignDistributionSet(ds1.getId(), savedTarget.getControllerId()).getAssignedEntity().iterator()
+                .next();
+        final Action deploymentAction1 = deploymentManagement
+                .findActiveActionsByTarget(PAGE, savedTarget.getControllerId()).getContent().get(0);
+        sendDeploymentActionFeedback(targetControllerId, deploymentAction1.getId(),
+                JsonBuilder.deploymentActionFeedback(deploymentAction1.getId().toString(), "closed", "failure", ""));
+        assertAttributesRequestStatus(targetControllerId, false);
+
+        savedTarget = assignDistributionSet(ds2.getId(), savedTarget.getControllerId()).getAssignedEntity().iterator()
+                .next();
+        final Action deploymentAction2 = deploymentManagement
+                .findActiveActionsByTarget(PAGE, savedTarget.getControllerId()).getContent().get(0);
+        sendDeploymentActionFeedback(targetControllerId, deploymentAction2.getId(),
+                JsonBuilder.deploymentActionFeedback(deploymentAction2.getId().toString(), "closed"));
+        assertAttributesRequestStatus(targetControllerId, true);
+    }
+
+    private void assertAttributesRequestStatus(final String targetControllerId, final boolean isRequested)
```

Comment:
Divide this method into two parts: assertThatAttributesUpdateIsRequested and assertThatAttributesUpdateIsNotRequested